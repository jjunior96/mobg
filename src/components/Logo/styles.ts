import styled, { css, DefaultTheme } from 'styled-components';

import { LogoProps } from '.';

type Props = Pick<LogoProps, 'size' | 'color'>;

export const Container = styled.div``;

const logoModifiers = {
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
  `,

  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
  `
};

export const Logo = styled.h2<Props>`
  ${({ theme, size, color }) => css`
    font-family: ${theme.font.family.secundary};
    font-weight: ${theme.font.bold};

    ${!!size && logoModifiers[size](theme)};
    color: ${theme.colors[color!]};
  `}
`;
