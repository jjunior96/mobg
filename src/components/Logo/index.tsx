import * as S from './styles';

export type LogoProps = {
  size?: 'medium' | 'large';
  color?: 'primary' | 'white' | 'black';
};

const Logo = ({ size = 'medium', color = 'primary' }: LogoProps) => {
  return (
    <S.Container>
      <S.Logo size={size} color={color}>
        mobg
      </S.Logo>
    </S.Container>
  );
};

export default Logo;
