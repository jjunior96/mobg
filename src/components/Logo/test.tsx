import { screen } from '@testing-library/react';

import { renderWithTheme } from '../../utils/tests/helpers';

import Logo from '.';

describe('<Logo />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Logo />);

    expect(screen.getByText(/mobg/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
