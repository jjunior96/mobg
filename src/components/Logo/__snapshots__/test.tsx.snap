// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the heading 1`] = `
.c0 {
  font-family: 'Lexend',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 600;
  font-size: 3.2rem;
  color: #5c63ef;
}

<div
  class=""
>
  <h2
    class="c0"
    color="primary"
  >
    mobg
  </h2>
</div>
`;
